internal/jptrace/translator.go

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package jptrace
+
+import (
+	jaegerTranslator "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/jaeger"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+
+	"github.com/jaegertracing/jaeger/model"
+)
+
+// ProtoFromTraces converts OpenTelemetry traces (ptrace.Traces)
+// to Jaeger model batches ([]*model.Batch).
+func ProtoFromTraces(traces ptrace.Traces) []*model.Batch {
+	batches := jaegerTranslator.ProtoFromTraces(traces)
+	spanMap := createSpanMapFromTraces(traces)
+	addWarningsToBatches(batches, spanMap)
+	return batches
+}
+
+func createSpanMapFromTraces(traces ptrace.Traces) map[pcommon.SpanID]ptrace.Span {
+	spanMap := make(map[pcommon.SpanID]ptrace.Span)
+	resources := traces.ResourceSpans()
+	for i := 0; i < resources.Len(); i++ {
+		scopes := resources.At(i).ScopeSpans()
+		for j := 0; j < scopes.Len(); j++ {
+			spans := scopes.At(j).Spans()
+			for k := 0; k < spans.Len(); k++ {
+				span := spans.At(k)
+				spanMap[span.SpanID()] = span
```

Comment:
I think it would be better the other way. You can't guarantee no dups in ptrace. If you keep a map for model.Span then you guarantee that all warnings from ptrace will be captured

Reply:
done